client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/CoapChannel.java

```diff
@@ -0,0 +1,265 @@
+package org.kaaproject.kaa.client.channel.impl.channels;
+
+import org.eclipse.californium.core.CoapClient;
+import org.eclipse.californium.core.CoapResponse;
+import org.eclipse.californium.core.Utils;
+import org.eclipse.californium.core.coap.MediaTypeRegistry;
+import org.kaaproject.kaa.client.channel.ChannelDirection;
+import org.kaaproject.kaa.client.channel.IpTransportInfo;
+import org.kaaproject.kaa.client.channel.KaaDataChannel;
+import org.kaaproject.kaa.client.channel.KaaDataDemultiplexer;
+import org.kaaproject.kaa.client.channel.KaaDataMultiplexer;
+import org.kaaproject.kaa.client.channel.ServerType;
+import org.kaaproject.kaa.client.channel.TransportConnectionInfo;
+import org.kaaproject.kaa.client.channel.TransportProtocolId;
+import org.kaaproject.kaa.client.channel.TransportProtocolIdConstants;
+
+import org.kaaproject.kaa.client.channel.connectivity.ConnectivityChecker;
+import org.kaaproject.kaa.common.TransportType;
+import org.kaaproject.kaa.common.endpoint.security.MessageEncoderDecoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: root
+ * Date: 10/24/16
+ * Time: 10:52 AM
+ * To change this template use File | Settings | File Templates.
+ */
+public class CoapChannel implements KaaDataChannel {
+
+  public static final Logger LOG = LoggerFactory // NOSONAR
+          .getLogger(CoapChannel.class);
+
+  private static final Map<TransportType, ChannelDirection> SUPPORTED_TYPES = new HashMap<TransportType, ChannelDirection>();
+
+  static {
+    SUPPORTED_TYPES.put(TransportType.PROFILE, ChannelDirection.BIDIRECTIONAL);
+    SUPPORTED_TYPES.put(TransportType.CONFIGURATION, ChannelDirection.BIDIRECTIONAL);
+    SUPPORTED_TYPES.put(TransportType.NOTIFICATION, ChannelDirection.BIDIRECTIONAL);
+  }
+
+  private static final String CHANNEL_ID = "default_coap_channel";
+  private KaaDataDemultiplexer demultiplexer;
+  private KaaDataMultiplexer multiplexer;
+  //    private volatile State channelState = State.CLOSED;
```

Comment:
Commented-out code is not allowed.

It's always a code smell, you know. Weeks after the next guys does a refactoring and wonders what's the purpose of commented out code, why it is commented, should it be uncommented or deleted?